client/src/components/ChallengeCard.tsx

```diff
@@ -0,0 +1,22 @@
+import Card from "react-bootstrap/Card";
+import { Button } from "react-bootstrap";
+
+type ChallengeCardProps = {
+  title: string;
+  description: string;
+  href: string;
+};
+
+function ChallengeCard({ title, description, href }: ChallengeCardProps) {
```

Comment:
> Are we using ids to identify the challenge as well?

yes, I have added IDs to each challenge. I think we will store them in the SQL database, so the ID will be the primary key.

Reply:
I'd rather keep them out of the ChallengeCard props since it's not explicitly required to display the card. A parent component (that fetches the challenges) can produce the href from the ids.

This pattern would result in better maintainability since the display and data are more decouples. Ex: we decide to change the database schema and only need to modify the data fetching element (which we should extract into a hook) rather than all of the views.

Reply:
ok sounds good I was just thinking if the ids were displayed like in leetcode it would be required. But if we aren't displaying ids thats fine